api4/team_test.go

```diff
@@ -622,24 +626,31 @@ func TestUpdateTeamPrivacy(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			team, resp := Client.UpdateTeamPrivacy(test.team.Id, test.privacy)
-			if test.errChecker != nil {
-				test.errChecker(t, resp)
-				return
-			} else {
-				CheckNoError(t, resp)
-				CheckOKStatus(t, resp)
-			}
-			require.Equal(t, test.wantType, team.Type)
-			require.Equal(t, test.wantOpenInvite, team.AllowOpenInvite)
-			if test.wantInviteIdChanged {
-				require.NotEqual(t, test.originalInviteId, team.InviteId)
-			} else {
-				require.Equal(t, test.originalInviteId, team.InviteId)
-			}
+			th.TestForAllClients(t, func(t *testing.T, client *model.Client4) {
+				team, resp := client.UpdateTeamPrivacy(test.team.Id, test.privacy)
+				if test.errChecker != nil {
+					test.errChecker(t, resp)
+					return
+				} else {
+					CheckNoError(t, resp)
+					CheckOKStatus(t, resp)
+				}
+				require.Equal(t, test.wantType, team.Type)
+				require.Equal(t, test.wantOpenInvite, team.AllowOpenInvite)
+				if test.wantInviteIdChanged {
+					require.NotEqual(t, test.originalInviteId, team.InviteId)
+				} else {
+					require.Equal(t, test.originalInviteId, team.InviteId)
+				}
+			})
 		})
 	}
 
+	t.Run("non-existent team", func(t *testing.T) {
+		_, resp := Client.UpdateTeamPrivacy(model.NewId(), model.TEAM_INVITE)
+		CheckForbiddenStatus(t, resp)
+	})
+
```

Comment:
Why split out as a separate test from the tests list above?

Reply:
I was going to ask the same question to @wiggin77 because this fails with 403 for non admin client but fails with 404 for admin users. Wonder if it would be OK to fail 404 on non-admin clients too.

Reply:
Strange, I would expect the opposite where a 403 is provided to admin client and 404 for users.  Discoverability of the team id should be reserved for admin users.

Reply:
If you want to add this to the table driven tests, you can add separate `wantXXX` fields for admin versus non-admin.

Reply:
Hmm, I see. I think this doesn't necessarily needs to be in the table tests. I'm 0/5 on adding another condition.

Reply:
I'm 0/5 on the implementation specifics but it would be good to cover all possible cases since the tests serve as proxy documentation.

Reply:
The permissions logic seems ok to me. 

The permission check is before any of the app operations, so a non-admin user gets a 403.
The admin user passes the permissions check but since the app can't find a team gets a 404.

Reply:
Some of the other APIs were changed to ensure 404 was returned on a permission fail to avoid discovery of valid user ids.  Not sure if channel ids need the same protection.  I'll approve and maybe that topic can be raised at a server meeting.

Reply:
> I'll approve and maybe that topic can be raised at a server meeting.

Sounds good to me.